call-notes/Epoch Rewards.md

```diff
@@ -0,0 +1,171 @@
+
+**Haron**: Yeah. But if you just care about the total number of rewards, I think you can still do it at the first block of the epoch boundary. 
+
+**Audience 2**: I can address a little bit of this. Since the rewards slot or blocks are gonna be paid out, and are gonna be deterministic, you're actually going to be able to generate a query based on which stake account or validator you want to know the rewards for. So we can actually make RPC queries that function the same. It will need some extension, but it'll be slightly different from what we have today.
+
+**Audience 1**: Per validator? And have to make, like, 2,000 queries or something. Is that what you're saying? 
+
+**Attendee 2**: No, no, I mean, you can just query all of the blocks in that case. Like, if you need to do a large range of rewards, you just get the entire payout or distribution block range for the slots. They're basically, gotta get blocks with a limit call with whatever the duration of the distribution.
+
+**Attendee 1**:  So basically what you have been watching has been reached when all the distribution is completed, and then make that call. Is that what you're suggesting? 
+
+
+**Attendee 3**: It’s deterministic, Yeah, you know, it's over by the thousandth block after the epoch, so you can just make the call right there instead, right?
+
+**Attendee 2**:  So today, where you just know that it's, you know, the first block in the epoch, now it's going to be, you know, the thousandth block, or that, you know, however many blocks before. I don't think any of this is fully parameterized yet. 
+
+**Attendee 1**: Okay, and so, well, so someone would have to have sort of code that understands that to be able to compute what that block would be, or there'll be an RPC call to ask that at some point. Like, is that how you anticipate it going? 
+
+
+**Attendee 3**: Yeah, basically. It'll depend on how much complexity there ends up being in the final thing. I don't anticipate it being that difficult. This could probably be like an RPC helper versus an actual RPC endpoint, but I think it'll be fairly trivial to put together a way to equivalently get, I mean, admittedly, it's going to be a little heavier because you have to get multiple blocks, right? 
+
+
+**Attendee 1** :  Okay. And I do have another question, but I don't want to monopolize this, so I'll wait to see if other people have a question first.
+
+**Jacob Creech**: All right, next question.
+
+I think it was Jeff, so go ahead, Jeffy, you should be able to unmute to ask your question. 
+
+**Jeff** : Oh, sorry, my mistake. 
+
+**Jacob Creech**: Okay, is there any other questions about this, or otherwise we'll go back to you, Zans. All right, Ashwin, go ahead.
+
+**Aswin**: Hey just wanted to ask, if there's a cluster we started in the first thousand blocks, is there any change to change that procedure because the snapshot won't have the result in it? 

```

Comment:
```suggestion
**Aswin**: Hey just wanted to ask, if there's a cluster restart in the first thousand blocks, is there any change to change that procedure because the snapshot won't have the result in it? 

```